arroyo/processing/strategies/dead_letter_queue/policies/produce.py

```diff
@@ -61,9 +64,17 @@ def _produce(self, payload: KafkaPayload) -> None:
         )
 
     def join(self, timeout: Optional[float] = None) -> None:
+        self.__producer.close()
```

Comment:
Please Close this after you have waited on all futures.
Now the producer may still be able to send you the callbacks after it was closed, but this is an implementation detail that may change. 
It is generally a safer option to close a resource when you are done using it.